test/lib/test-collector.js

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+// This provides an in-process http server to use in place of
+// collector.newrelic.com. It allows for custom handlers so that test specific
+// assertions can be made.
+
+const https = require('node:https')
+const querystring = require('node:querystring')
+const helper = require('./agent_helper')
+const fakeCert = require('./fake-cert')
+
+class Collector {
```

Comment:
This should be generic enough to utilize in all tests that need to talk to the remote collector. I haven't actually looked, but I bet we can replace the `fake-collector` references in `test/integration/collector-remote-method.tap.js` with this.